services/llgen/kilda/llgen/manager.py

```diff
@@ -0,0 +1,163 @@
+import errno
+import json
+import logging.config
+import os
+import socket
+import sys
+
+from kilda.llgen import context as llgen_context
+from kilda.llgen import daemon as daemon_module
+from kilda.llgen import protocol
+from kilda.llgen import traffic
+from kilda.llgen import utils
+
+
```

Comment:
This class needs some comments as well. How is it used? Why does it exist? Anything interesting about the algorithm? Any sub-optimal areas?